Taskfile.yml

```diff
@@ -177,14 +181,15 @@ tasks:
     - "controllers/**/*.go"
     - "pkg/**/*.go"
     cmds:
-    - go build -ldflags "-X {{.CONTROLLER_PACKAGE}}/pkg/version.BuildVersion={{.VERSION}}" -o ./bin/{{.CONTROLLER_APP}}
+    - CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-X {{.CONTROLLER_PACKAGE}}/pkg/version.BuildVersion={{.VERSION}}" -o ./bin/{{.CONTROLLER_APP}}
```

Comment:
Do we absolutely need to nail down the `OS` and `ARCH` like this?

Reply:
I think we want to set them. According to [this](https://www.digitalocean.com/community/tutorials/building-go-applications-for-different-operating-systems-and-architectures), `GOOS` and `GOARCH` are used to set the _target_ OS and Arch. Since the intent of this binary is that it be used and run inside of the container built in the `controller:docker-build` step, which is a Linux container, I think it makes sense to force Linux here so that the result of this step will always work with the `controller:docker-build` step.

This command should work just fine on Windows - it will just produce a binary targeting Linux (which is what we want AFAIK).

Reply:
Given that the binary is going to be used inside the container, this makes sense. :+1: